botocore/client.py

```diff
@@ -655,3 +655,32 @@ def _validate_s3_configuration(self, s3):
             if addressing_style not in ['virtual', 'auto', 'path', None]:
                 raise InvalidS3AddressingStyleError(
                     s3_addressing_style=addressing_style)
+
+    def merge(self, other_config):
+        """Merges the config object with another config object
+
+        This will merge in all non-default values from the provided config
+        and return a new config object
+
+        :type other_config: botocore.client.Config
+        :param other config: Another config object to merge with. The values
+            in the provided config object will take precedence in the merging
+
+        :rtype: botocore.client.Config
+        :returns: A config object built from the merged values of both
+            config objects.
+        """
+        # Make a copy of the current attributes in the config object.
+        config_properties = copy.copy(vars(self))
+
+        # Merge the config object's attributes with the other config
+        # object's attributes.
+        for property_key, property_value in vars(other_config).items():
+            if property_key in ['connect_timeout', 'read_timeout']:
```

Comment:
What would be generic? In terms of searching for timeout in the name instead of searching for if the key is in the list? Or how all config values are handled? The most general logic I could have is:
```py
for property_key, property_value in vars(other_config).items():
    if property_value not in [None, DEFAULT_TIMEOUT]:
        config_properties[property_key] = property_value
```
But that pretty much black lists any config option that uses the value ``60``. Given none of the other config options use integers, I think I would work for now. I was just trying to prevent a possible future bug if an integer property gets added to the config later on.